src/compile/scale.ts

```diff
@@ -239,42 +241,43 @@ export function domain(scale: Scale, model: Model, channel:Channel): any {
   sort = domainSort(model, channel, scale.type);
 
   if (includeRawDomain) { // includeRawDomain - only Q/T
+    model.component.data.includeRawDomain = true;
     return {
-      data: SOURCE,
+      data: SCALE,
       field: model.field(channel, {noAggregate: true})
     };
   } else if (fieldDef.bin) { // bin
     return scale.type === ScaleType.ORDINAL ? {
       // ordinal bin scale takes domain from bin_range, ordered by bin_start
-      data: model.dataTable(),
+      data: model.dataName(SCALE),
       field: model.field(channel, { binSuffix: '_range' }),
       sort: {
         field: model.field(channel, { binSuffix: '_start' }),
         op: 'min' // min or max doesn't matter since same _range would have the same _start
       }
     } : channel === COLOR ? {
       // Currently, binned on color uses linear scale and thus use _start point
-      data: model.dataTable(),
+      data: model.dataName(SCALE),
       field: model.field(channel, { binSuffix: '_start' })
     } : {
       // other linear bin scale merges both bin_start and bin_end for non-ordinal scale
-      data: model.dataTable(),
+      data: model.dataName(SCALE),
       field: [
         model.field(channel, { binSuffix: '_start' }),
         model.field(channel, { binSuffix: '_end' })
       ]
     };
   } else if (sort) { // have sort -- only for ordinal
     return {
-      // If sort by aggregation of a specified sort field, we need to use SOURCE table,
+      // If sort by aggregation of a specified sort field, we need to use SCALE table,
       // so we can aggregate values for the scale independently from the main aggregation.
-      data: sort.op ? SOURCE : model.dataTable(),
+      data: sort.op ? SCALE : model.dataName(SCALE),
```

Comment:
why should `model.dataName(SCALE)` return something different from `SCALE` in this case?